lib/core/src/msParam.cpp

```diff
@@ -1022,6 +1022,28 @@ parseMspForPosInt( msParam_t* inpParam ) {
     return myInt;
 }
 
+int parseMspForStr_new(msParam_t* inpParam, char** outParam)
+{
+    if (inpParam == NULL || inpParam->inOutStruct == NULL) {
+        rodsLog(LOG_ERROR, "%s: inpParam type %s is not STR_MS_T", __func__, inpParam->type);
+        *outParam = NULL;
+        return 1;
+    }
+
+    if (strcmp(inpParam->type, STR_MS_T) != 0) {
+        rodsLog(LOG_ERROR, "%s: inpParam type %s is not STR_MS_T", __func__, inpParam->type);
+        return 1;
+    }
+
+    if (strcmp((char*) inpParam->inOutStruct, "null") == 0) {
```

Comment:
Prefer the C++ casts rather than the C casts.

_This applies to the other locations too._